daemon/delete.go

```diff
@@ -77,6 +77,22 @@ func (daemon *Daemon) rmLink(container *container.Container, name string) error
 // cleanupContainer unregisters a container from the daemon, stops stats
 // collection and cleanly removes contents and metadata from the filesystem.
 func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemove, removeVolume bool) (err error) {
+	// If force removal is required, delete container from various
+	// indexes even if removal failed.
+	defer func() {
+		if err == nil || forceRemove {
```

Comment:
I'm not sure I understand the requested change.The following statement in the function checks for an error and ignores it with only a message printed in the logs:

```go
if err := container.ToDiskLocking(); err != nil && !os.IsNotExist(err) {
    logrus.Errorf("Error saving dying container to disk: %v", err)
}   
```

If I change that to `if err = container.ToDiskLocking();` then the outer scope `err` will be set to this value which means that it will be able to affect the output of the function. Is this the intended change of semantics?